injector/network_disruption.go

```diff
@@ -53,6 +57,34 @@ type NetworkDisruptionInjectorConfig struct {
 	TrafficController network.TrafficController
 	NetlinkAdapter    network.NetlinkAdapter
 	DNSClient         network.DNSClient
+	State             DisruptionState
+}
+
+type DisruptionState struct {
+	State chan InjectorState
+}
+
+// tcServiceFilter describes a tc filter, representing the service filtered and its preferences
+type tcServiceFilter struct {
+	service     networkDisruptionService
+	preferences map[string][]string // identifiers used to delete the filter, a map of interfaces and preferences
```

Comment:
Switched the logic to using only one preference (also called priority) per service. It was easier to assign directly the priority at the add filter level so that the logic to discover the priority can be removed.